internal/resources/cluster/dataprotection/datasource_enable_data_protection.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package dataprotection
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	dataprotectionmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/dataprotection"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+)
+
+func DataSourceEnableDataProtection() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceEnableDataProtectionRead,
+		Schema:      enableDataProtectionSchema,
+	}
+}
+
+func dataSourceEnableDataProtectionRead(ctx context.Context, data *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	var (
+		resp *dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionListDataProtectionsResponse
+		err  error
+	)
+
+	responseStatus := dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionStatusPhasePHASEUNSPECIFIED
+
+	for retries := 0; retries < 50; retries++ {
+		resp, err = config.TMCConnection.DataProtectionService.DataProtectionResourceServiceList(data.Get(clusterNameKey).(string))
+
+		if err != nil || resp == nil {
+			return diag.Errorf("Couldn't receive data protection configurations for cluster: %s", data.Get(clusterNameKey))
+		}
+
+		dataProtection := resp.DataProtections[0]
+		responseStatus = *dataProtection.Status.Phase
+
+		if responseStatus != dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionStatusPhaseREADY {
+			time.Sleep(5 * time.Second)
+		} else {
+			break
+		}
+	}
+
+	if responseStatus != dataprotectionmodels.VmwareTanzuManageV1alpha1ClusterDataprotectionStatusPhaseREADY {
+		return diag.Errorf("Couldn't receive data protection configurations for cluster: %s (Retries exceeded)", data.Get(clusterNameKey))
+	}
+
+	if len(resp.DataProtections) == 0 {
+		diags = diag.Errorf("Couldn't find data protection configuration for cluster: %s", data.Get(clusterNameKey))
+	} else {
+		var (
+			schemaDisableRestic   interface{}
+			schemaBackupLocations interface{}
+			dataProtection        = resp.DataProtections[0]
+			oldSpec               = data.Get(specKey).([]interface{})[0].(map[string]interface{})
+		)
+
+		if len(oldSpec) > 0 {
+			// Disable restic doesn't return from API.
+			schemaDisableRestic = oldSpec[disableResticKey]
+
+			// Backup location might return with values when not configured explicitly.
+			schemaBackupLocations = oldSpec[backupLocationNamesKey]
+		}
+
+		err = tfModelConverter.FillTFSchema(dataProtection, data)
+
+		if err != nil {
+			diags = diag.FromErr(err)
+		}
+
+		data.SetId(dataProtection.FullName.ClusterName)
```

Comment:
tbh I don't understand this code well but I guess this is the unique identifier for the resource. Cluster name is insufficient for this since a cluster is identified uniquely by (org id, management cluster name, provisioner name, cluster name). Could alternatively use the cluster id or the data protection resource uid (should be under meta.uid)

Reply:
Just tell me what would you prefer to be set as the resource UID in the TF state and I shall change it

Reply:
@shobha2626 what is the convention for cluster scoped resources? I see 3 options:
- cluster full name + resource name
- cluster uid + resource name
- resource uid
cc @eastaga

Reply:
@ramya-bangera do you have any recommendation for this?

Reply:
resource ID should be a unique identifier, setting it to any such unique identifier should be good.